src/keep_gpu/single_gpu_controller/cuda_gpu_controller.py

```
@@ -269,23 +289,19 @@ def _keep_loop(
                 if "out of memory" in str(e).lower():
+                    confirm_startup()
                     if stop_evt.wait(self.interval):
                         return
                     continue
-                self._failure_exc = RuntimeError(
-                    f"rank {self.rank}: unexpected CUDA keep worker failure: {e}"
-                )
-                logger.exception("%s", self._failure_exc)
+                record_worker_failure(e)
                 return
             except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)